File: abc181c.py

import math

N = int(input()) 
lists = []
dist_lists = []
for i in range(N):
    x, y = map(int,input().split())     
    lists.append([x,y])

for i in range(N):
    for j in range(N):
        if i != j:
            dist = round(math.sqrt((lists[i][0] - lists[j][0]) ** 2 + (lists[i][1] - lists[j][1]) ** 2), 4)
            dist_lists.append(dist)

m = len(dist_lists)

ans = "No"
for i in range(m):
    for j in range(m):
        for k in range(m):
            if i!= j and j != k and k != i:
                if dist_lists[i] == dist_lists[j] + dist_lists[k]:
                    ans = "Yes"
                    break
print(ans)

File: abc181c.cpp


#include <bits/stdc++.h>
using namespace std;


int main(void) {
    int N;
    cin >> N;

    vector< vector<int> > data(N, vector<int>(2));

    for (int i = 0; i < N; i++) {
        int x,y;
        cin >> x >> y;    
        data.at(i).at(0) = x;
        data.at(i).at(1) = y;
        }
    
    int n = (N * (N - 1)) / 2;
    vector<int> distance;
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (i != j) {
                int x_diff = data.at(i).at(0) - data.at(j).at(0);
                int y_diff = data.at(i).at(1) - data.at(j).at(1);
                int dist = sqrt((x_diff * x_diff) + (y_diff * y_diff));
                distance.push_back(dist);
                //cout << distance.at(i) << endl;
            }
        }
    }
    string ans = "No";
    for (int i = 0; i < n; i++) {
        for (int j = 0;j < n; j++) {
            for (int k = 0;k < n; k++) {
                if (i != j && j != k && k != i) {
                    if (distance.at(i) == distance.at(j) + distance.at(k)) {
                        ans = "Yes";
                        break;
                        }
                    }
                }
            }
        }
    cout << ans << endl;
    return 0;
    }
